lib/thor/actions/file_manipulation.rb

```diff
@@ -260,6 +261,32 @@ def inject_into_module(path, module_name, *args, &block)
     #     match << " no more. Use thor!"
     #   end
     #
+    def gsub_file!(path, flag, *args, &block)
+      config = args.last.is_a?(Hash) ? args.pop : {}
+
+      config[:error_on_no_change] = true
```

Comment:
I'm ok with this feature but I don't think we should add a config to `gsub_file` to implement this. I think we can implement this by moving the logic to a private method that we can then add the option, or just use a block to change the behavior using the template method pattern.

Reply:
I've pushed up a change that is hopefully in the right direction - let me know what you think.

At this point I personally feel having the option on `gsub_file` is better because it's more flexible and feels like the DRYest way to do it already, but `gsub_file!` is most important to me so happy to go with your preference (and maybe I'm missing something about what you mean - I'm not sure what the "template method pattern" is at this point).